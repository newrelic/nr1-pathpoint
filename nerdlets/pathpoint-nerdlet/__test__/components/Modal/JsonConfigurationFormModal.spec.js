import { shallow } from 'enzyme';
import React from 'react';
import { create } from 'react-test-renderer';
import {
  BodyJsonConfigurationFormModal,
  HeaderJsonConfigurationFormModal
} from '../../../components/Modal/JsonConfigurationFormModal';

describe('JsonConfigurationFormModal component', () => {
  test('Render body with default data', () => {
    const body = create(
      <BodyJsonConfigurationFormModal
        _onClose={jest.fn()}
        GetCurrentConfigurationJSON={jest.fn()}
        SetConfigurationJSON={jest.fn()}
        validateKpiQuery={{}}
        UpdateJSONMetaData={jest.fn()}
        jsonMetaData={{
          description: '',
          note: ''
        }}
        GetHistoricJSONData={jest.fn().mockReturnValue([])}
        JSONModal={{
          view: 0,
          historic: []
        }}
        UpdateItemSelectFromHistoric={jest.fn()}
        currentHistoricSelected={null}
        RestoreJSONFromHistoric={jest.fn()}
        username="PathPoint"
      />
    );
    expect(body.toJSON()).toMatchSnapshot();
  });

  test('Render body with default data and do functions', () => {
    const onClose = jest.fn().mockReturnValue(true);
    const validateKpiQuery = { query: 'SELECT *FROM Transaction' };
    const SetConfigurationJSON = jest.fn().mockReturnValue(true);
    const component = shallow(
      <BodyJsonConfigurationFormModal
        _onClose={onClose}
        GetCurrentConfigurationJSON={jest.fn()}
        SetConfigurationJSON={SetConfigurationJSON}
        validateKpiQuery={validateKpiQuery}
        UpdateJSONMetaData={jest.fn()}
        jsonMetaData={{
          description: '',
          note: ''
        }}
        GetHistoricJSONData={jest.fn().mockReturnValue([])}
        JSONModal={{
          view: 0,
          historic: []
        }}
        UpdateItemSelectFromHistoric={jest.fn()}
        currentHistoricSelected={null}
        RestoreJSONFromHistoric={jest.fn()}
        username="PathPoint"
      />
    );
    expect(component.exists()).toBe(true);
  });

  test('Render header with default data', () => {
    const header = create(<HeaderJsonConfigurationFormModal />);
    expect(header.toJSON()).toMatchSnapshot();
  });
});
