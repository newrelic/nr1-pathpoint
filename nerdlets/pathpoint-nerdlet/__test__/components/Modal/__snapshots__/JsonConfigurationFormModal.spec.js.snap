// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JsonConfigurationFormModal component Render body with default data 1`] = `
<div
  style={
    Object {
      "width": "400px",
    }
  }
>
  <div>
    <div
      style={
        Object {
          "marginBottom": "20px",
        }
      }
    >
      <label>
        Description: 
      </label>
      <input
        className="text-input-custom"
        maxLength={64}
        onChange={[Function]}
        type="text"
      />
    </div>
    <div>
      <label>
        Note: 
      </label>
      <textarea
        className="text-input-custom"
        onChange={[Function]}
        rows="5"
        type="text"
      />
    </div>
    <div
      style={
        Object {
          "marginTop": "10px",
        }
      }
    >
      <p>
        User:
        <strong>
           PathPoint
        </strong>
      </p>
    </div>
  </div>
  <div
    id="upload-file-submit"
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
        "marginTop": "30px",
      }
    }
  >
    <button
      className="btn btn-default"
      color="primary"
      disabled={true}
      onClick={[Function]}
      style={
        Object {
          "background": "#0178bf",
          "color": "white",
          "marginRight": "20px",
        }
      }
      type="submit"
      variant="contained"
    >
      <img
        src={Object {}}
        style={
          Object {
            "height": "auto",
            "marginRight": "4px",
            "width": "16px",
          }
        }
      />
      From File
    </button>
    <div
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <button
        className="btn btn-default"
        color="primary"
        disabled={false}
        onClick={[MockFunction]}
        style={
          Object {
            "background": "#0178bf",
            "color": "white",
          }
        }
        type="button"
        variant="contained"
      >
        <img
          src={Object {}}
          style={
            Object {
              "marginRight": "4px",
            }
          }
        />
        From History
      </button>
    </div>
  </div>
  <input
    accept=".json"
    id="file-upload"
    onChange={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    type="file"
  />
</div>
`;

exports[`JsonConfigurationFormModal component Render header with default data 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "width": "100%",
    }
  }
>
  <div
    className="titleModal"
    style={
      Object {
        "width": "50%",
      }
    }
  >
    <img
      src={Object {}}
      width="18"
    />
     Json Config.
    <img
      onClick={[Function]}
      src={Object {}}
      style={
        Object {
          "cursor": "pointer",
          "marginLeft": "10px",
        }
      }
    />
  </div>
  <div
    className="container_header_icons"
  >
    <div
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "height": "25px",
          "marginLeft": "20px",
          "position": "relative",
          "width": "25px",
        }
      }
    >
      <svg
        fill="#017C86"
        focusable="false"
        height="20"
        role="img"
        style={
          Object {
            "height": "100%",
            "left": "0px",
            "position": "absolute",
            "top": "0px",
            "width": "100%",
          }
        }
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.4 8.4l-.8-.8L8 10.3V1H7v9.3L4.4 7.6l-.8.8 3.9 3.8z"
        />
        <path
          d="M13 10v4H2v-4H1v5h13v-5z"
        />
      </svg>
      <div
        style={
          Object {
            "left": "0px",
            "position": "absolute",
            "top": "-1000px",
          }
        }
      >
        <a
          className="downloadLink"
          href="#"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          Pathpoint_Json_v1.5
        </a>
      </div>
    </div>
  </div>
</div>
`;
